### ISSUE TITLE
v1.8.1 fails to build with gcc7

### ISSUE THREAD

**Issue by xnox** [libavg/libavg#793]
from https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=853481

```
In file included from GLShaderParam.h:31:0,
                 from GPUHueSatFilter.h:29,
                 from GPUHueSatFilter.cpp:22:
GLShaderParam.h: In member function 'void avg::GLShaderParamTemplate<VAL_TYPE>::uniformSet(unsigned int, const VAL_TYPE&)':
../base/Exception.h:84:87: error: cannot convert 'std::__cxx11::basic_string<char>' to 'const char*' for argument '4' to 'void avg::avgAssert(bool, const char*, int, const char*)'
 #define AVG_ASSERT_MSG(b, pszReason) avgAssert((b) != 0, __FILE__, __LINE__, pszReason);
                                                                                       ^
GLShaderParam.h:81:9: note: in expansion of macro 'AVG_ASSERT_MSG'
         AVG_ASSERT_MSG(false,
         ^~~~~~~~~~~~~~
In file included from GLShaderParam.h:31:0,
                 from OGLShader.h:27,
                 from StandardShader.h:27,
                 from StandardShader.cpp:22:
GLShaderParam.h: In member function 'void avg::GLShaderParamTemplate<VAL_TYPE>::uniformSet(unsigned int, const VAL_TYPE&)':
../base/Exception.h:84:87: error: cannot convert 'std::__cxx11::basic_string<char>' to 'const char*' for argument '4' to 'void avg::avgAssert(bool, const char*, int, const char*)'
 #define AVG_ASSERT_MSG(b, pszReason) avgAssert((b) != 0, __FILE__, __LINE__, pszReason);
                                                                                       ^
GLShaderParam.h:81:9: note: in expansion of macro 'AVG_ASSERT_MSG'
         AVG_ASSERT_MSG(false,
         ^~~~~~~~~~~~~~
Makefile:1031: recipe for target 'GPUInvertFilter.lo' failed
make[5]: *** [GPUInvertFilter.lo] Error 1
Makefile:1031: recipe for target 'OGLShader.lo' failed
make[5]: *** [OGLShader.lo] Error 1
Makefile:1031: recipe for target 'ShaderRegistry.lo' failed
make[5]: *** [ShaderRegistry.lo] Error 1
Makefile:1031: recipe for target 'GLContext.lo' failed
make[5]: *** [GLContext.lo] Error 1
Makefile:1031: recipe for target 'GPUHueSatFilter.lo' failed
make[5]: *** [GPUHueSatFilter.lo] Error 1
Makefile:1031: recipe for target 'StandardShader.lo' failed
make[5]: *** [StandardShader.lo] Error 1
make[5]: Leaving directory '/<<PKGBUILDDIR>>/src/graphics'
Makefile:1046: recipe for target 'all-recursive' failed
make[4]: *** [all-recursive] Error 1
make[4]: Leaving directory '/<<PKGBUILDDIR>>/src/graphics'
Makefile:453: recipe for target 'all-recursive' failed
make[3]: *** [all-recursive] Error 1
make[3]: Leaving directory '/<<PKGBUILDDIR>>/src'
Makefile:392: recipe for target 'all' failed
make[2]: *** [all] Error 2
make[2]: Leaving directory '/<<PKGBUILDDIR>>/src'
Makefile:460: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/<<PKGBUILDDIR>>'
dh_auto_build: make -j64 returned exit code 2
debian/rules:4: recipe for target 'build' failed
make: *** [build] Error 2
dpkg-buildpackage: error: debian/rules build gave error exit status 2
```

Does master branch builds with gcc7? Will there be a new release with gcc7 compat?

**Comment by scotty007** [comment]
I just tried to build the libavg master branch on debian-testing with gcc7 and it failed with the same error.

**Comment by scotty007** [comment]
I started a [branch](https://github.com/libavg/libavg/tree/features/gcc6_gcc7) to fix gcc6/gcc7 builds. The above issue is solved. Another one appeared (missing ```-lpython2.7``` linker flag). Working on it ...

**Comment by scotty007** [comment]
The broken linking (see above) seems to be debian specific. On ubuntu 17.04 everything runs fine, on debian testing and 9.1.0 the build fails when linking the test* executables (several ```libboost_python.so: undefined reference to `Py...` ```). 17.04 and 9.1.0 provide the same versions of cmake (3.7.2), gcc (6.3.0) and libboost-python (1.62.0). The invoked linker command is exactly the same on both systems.

To get the build running on debian, I manuallly set ```CMAKE_EXE_LINKER_FLAGS=-lpython2.7```.

**Comment by scotty007** [comment]
Fixed in 2ed5eb3.
